packages/compiler/src/templates/Panels.svelte

```diff
@@ -0,0 +1,37 @@
+<script>
+  import { onMount } from "svelte";
+  const contents = "Test\n---\nTest2";
+  export let value = "";
+  let cards = [];
+  let reactiveCards = [];
+
+  $: console.log(value);
+
+  onMount(() => {
+    if (contents !== "") {
+      cards = parsePanels(value);
+    }
+  })
+
+  function parsePanels (contents) {
+    const panelDelimiterRegex = /\n\-{3,}\n/;
+    const cards = contents.split(panelDelimiterRegex);
+    return cards;
+  }
```

Comment:
I think we should delegate this type of logic to the compiler. I'd probably pass the content into this component as *named slots* which we can then arrange inside this component:

https://svelte.dev/tutorial/named-slots

It might be easiest to prototype that logic inside the svelte repl.

Reply:
Oh neat; yeah a named slot makes a lot more sense, and I agree with moving the kind of regex required for panels to the compiler. For panels in particular, there are other regex sequences based on optional headers (```^^^```) and footers (```+++```), so it would make sense to be able to directly pass the header and footer contents to the named slots.